src/osd/ReplicatedBackend.cc

```diff
@@ -1295,6 +1295,14 @@ void ReplicatedBackend::calc_head_subsets(
   uint64_t size = obc->obs.oi.size;
   if (size)
     data_subset.insert(0, size);
+  if (parent->min_peer_features() & CEPH_OSD_PARTIAL_RECOVERY) {
+    const auto it = missing.get_items().find(head);
+    assert(it != missing.get_items().end());
+    data_subset.intersection_of(it->second.clean_regions.get_dirty_regions());
+    dout(10) << "calc_head_subsets " << head
+             << "  data_subset " << data_subset << dendl;
+    return;
```

Comment:
@athanatos  If the head object is modified, it implicitly indicates that the head content is different from the snapshot content.  So there is no chance to copy the unmodified content from the snapshot.  Furthermore, I think we should replace the clone_overlap in snapset as bounded_lossy_interval_set as you mentioned before? Currently, we find that taking snapshot sometimes makes too many item in clone_overlap, which lead to a quite large snapset attr (this will be ruined the performance during rebuiding ObjectContext )